### ISSUE TITLE
Minesweep compile error on Cygwin

### ISSUE THREAD

**Issue by Griffinhart** [CleverRaven/Cataclysm-DDA#14124]
e:
`#define VERSION "0.C-7602-g931dfcb"`

```
g++  -DCATA_NO_CPP11_STRING_CONVERSIONS -DLOCALIZE -ffast-math -Wall -Wextra -g  --std=c++11 -MMD -D_XOPEN_SOURCE=600 -c src/iuse_software_minesweeper.cpp -o obj/iuse_software_minesweeper.o
src/iuse_software_minesweeper.cpp: In lambda function:
src/iuse_software_minesweeper.cpp:49:20: error: ‘atoi’ is not a member of ‘std’
             iVal = std::atoi(string_input_popup(sType.c_str(), 5, std::to_string(iVal), ssTemp.str().c_str(), "", -1, true).c_str());
                    ^
src/iuse_software_minesweeper.cpp:49:20: note: suggested alternative:
In file included from /usr/include/wchar.h:4:0,
                 from /usr/lib/gcc/i686-pc-cygwin/4.9.3/include/c++/cwchar:44,
                 from /usr/lib/gcc/i686-pc-cygwin/4.9.3/include/c++/bits/postypes.h:40,
                 from /usr/lib/gcc/i686-pc-cygwin/4.9.3/include/c++/bits/char_traits.h:40,
                 from /usr/lib/gcc/i686-pc-cygwin/4.9.3/include/c++/string:40,
                 from src/iuse_software_minesweeper.cpp:1:
/usr/include/stdlib.h:74:5: note:   ‘atoi’
 int _EXFUN(atoi,(const char *__nptr));
     ^
src/iuse_software_minesweeper.cpp:49:67: error: ‘to_string’ is not a member of ‘std’
             iVal = std::atoi(string_input_popup(sType.c_str(), 5, std::to_string(iVal), ssTemp.str().c_str(), "", -1, true).c_str());
                                                                   ^
src/iuse_software_minesweeper.cpp:49:67: note: suggested alternative:
In file included from src/rng.h:4:0,
                 from src/iuse_software_minesweeper.cpp:11:
src/compatibility.h:56:20: note:   ‘to_string’
 inline std::string to_string( double const n )
                    ^
src/iuse_software_minesweeper.cpp: In lambda function:
src/iuse_software_minesweeper.cpp:225:26: error: ‘to_string’ is not a member of ‘std’
                          std::to_string(mLevel[y][x]));
                          ^
src/iuse_software_minesweeper.cpp:225:26: note: suggested alternative:
In file included from src/rng.h:4:0,
                 from src/iuse_software_minesweeper.cpp:11:
src/compatibility.h:56:20: note:   ‘to_string’
 inline std::string to_string( double const n )
                    ^
src/iuse_software_minesweeper.cpp: In member function ‘int minesweeper_game::start_game()’:
src/iuse_software_minesweeper.cpp:270:38: error: ‘to_string’ is not a member of ‘std’
                             sGlyph = std::to_string(mLevel[iPlayerY][iPlayerX]);
                                      ^
src/iuse_software_minesweeper.cpp:270:38: note: suggested alternative:
In file included from src/rng.h:4:0,
                 from src/iuse_software_minesweeper.cpp:11:
src/compatibility.h:56:20: note:   ‘to_string’
 inline std::string to_string( double const n )
                    ^
Makefile:489: recipe for target 'obj/iuse_software_minesweeper.o' failed
make: *** [obj/iuse_software_minesweeper.o] Error 1
```

Not sure about the `std::atoi` error, but it looks like code is calling `std::to_string` instead of the `to_string` declared in `compatibility.h`.

**Comment by DanmakuDan** [comment]
Cygwin apparently has incomplete functionality related to vswprintf (_GLIBCXX_HAVE_BROKEN_VSWPRINTF flag disables std::atoi etc, std::tostring). You can manually modify the GCC headers to remove the block on the missing functions, but otherwise, it appears to be a Cygwin limitation (going on 5 years?). 

http://stackoverflow.com/questions/8542221/stdstoi-doesnt-exist-in-g-4-6-1-on-mingw

**Comment by illi-kun** [comment]
Can be closed since #14111 is merged.
